File: src/main/java/com/baidu/algorithm/array/_419_Battleships_in_a_Board.py
class Solution(object):
    def countBattleships(self, board):
        """
        :type board: List[List[str]]
        :rtype: int
        """
        if not board or not board[0]:
            return 0

        m, n, res = len(board), len(board[0]), 0
        for i in xrange(m):
            for j in xrange(n):
                if board[i][j] == 'X' and (i == 0 or board[i - 1][j] == '.') and (j == 0 or board[i][j - 1] == '.'):
                    res += 1

        return res

File: src/main/java/com/baidu/algorithm/array/_419_Battleships_in_a_Board.java
/*
 * Copyright (C) 2018 Baidu, Inc. All Rights Reserved.
 */
package com.baidu.algorithm.array;

/**
 * _419_Battleships_in_a_Board
 *
 * @author xuhaoran01
 */
public class _419_Battleships_in_a_Board {

    public int countBattleships(char[][] board) {
        if (board == null || board.length == 0) {
            return 0;
        }

        int res = 0, m = board.length, n = board[0].length;
        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                if (board[i][j] == '.') {
                    continue;
                } else if (i > 0 && board[i - 1][j] == 'X') {
                    continue;
                } else if (j > 0 && board[i][j - 1] == 'X') {
                    continue;
                } else {
                    res++;
                }
            }
        }

        return res;
    }
}
